controls_rewrite_folder/driver_interface_error_handling.cpp

```diff
@@ -0,0 +1,47 @@
+#include <iostream>
+#include <cmath>
+//everson shen and aleeza ali zar
+
+
+
+double potentiometerProcessing(double fractionIn){
+
+    if (fractionIn >= 0 && fractionIn <= 1){
+        return fractionIn;
+    }
+    else{
+        return 0;
+    }
+}
+
+bool b_DriverInterfaceError(double DI_V_AccelPedalPos1, double DI_V_AccelPedalPos2, double DI_FrontBrakePressure, double DI_RearBrakePressure, double DI_V_RawSteeringAngle){
```

Comment:
Great observation about the `double` datatype. We can "template" our functions to work on any datatype. Can you update both functions to use templates? https://en.cppreference.com/w/cpp/language/templates